File: shared/utils/env_validator.ts
/**
 * Centralized environment variable validation utility for TypeScript services.
 * Import and call validateRequiredEnvVars() at service startup.
 */

export interface EnvValidationOptions {
  serviceName?: string;
  exitOnError?: boolean;
}

/**
 * Validate that all required environment variables are set.
 * Exits with code 1 if any are missing (unless exitOnError is false).
 */
export function validateRequiredEnvVars(
  requiredVars: string[],
  options: EnvValidationOptions = {}
): boolean {
  const { serviceName, exitOnError = true } = options;
  const missing: string[] = [];

  for (const varName of requiredVars) {
    if (!process.env[varName]) {
      missing.push(varName);
    }
  }

  if (missing.length > 0) {
    const servicePrefix = serviceName ? `[${serviceName}] ` : '';
    console.error(`FATAL: ${servicePrefix}Missing required environment variables:`);
    missing.forEach(varName => {
      console.error(`  - ${varName}`);
    });
    console.error('\nPlease set these variables in your .env file or environment.');
    
    if (exitOnError) {
      process.exit(1);
    }
    return false;
  }

  return true;
}

/**
 * Get environment variable or exit if not set.
 */
export function getEnvOrExit(varName: string, serviceName?: string): string {
  const value = process.env[varName];
  if (!value) {
    const servicePrefix = serviceName ? `[${serviceName}] ` : '';
    console.error(`FATAL: ${servicePrefix}Missing required environment variable: ${varName}`);
    process.exit(1);
  }
  return value;
}

File: shared/utils/env_validator.py
"""
Centralized environment variable validation utility for Python services.
Import and call validate_required_env_vars() at service startup.
"""
import os
import sys
from typing import List, Optional


def validate_required_env_vars(
    required_vars: List[str],
    service_name: Optional[str] = None
) -> None:
    """
    Validate that all required environment variables are set.
    Exits with code 1 if any are missing.
    
    Args:
        required_vars: List of required environment variable names
        service_name: Optional service name for error messages
    """
    missing = [var for var in required_vars if not os.getenv(var)]
    
    if missing:
        service_prefix = f"[{service_name}] " if service_name else ""
        print(f"FATAL: {service_prefix}Missing required environment variables:", file=sys.stderr)
        for var in missing:
            print(f"  - {var}", file=sys.stderr)
        print("\nPlease set these variables in your .env file or environment.", file=sys.stderr)
        sys.exit(1)


def get_env_or_exit(var_name: str, service_name: Optional[str] = None) -> str:
    """
    Get environment variable or exit if not set.
    
    Args:
        var_name: Environment variable name
        service_name: Optional service name for error messages
        
    Returns:
        Environment variable value
    """
    value = os.getenv(var_name)
    if not value:
        service_prefix = f"[{service_name}] " if service_name else ""
        print(f"FATAL: {service_prefix}Missing required environment variable: {var_name}", file=sys.stderr)
        sys.exit(1)
    return value
